hack/git_client_side_hooks/pre-push

```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+
+# Copyright 2022 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# this script will verify signoff checking before the push is successful.
+# Information about the commits which are being pushed is supplied as lines to
+# the standard input in the form:
+#
+#   <LOCAL_REF> <LOCAL_SHA> <REMOTE_REF> <REMOTE_SHA>
+
```

Comment:
Probably we can add the reference link here for more details: https://www.git-scm.com/docs/githooks#_pre_push